src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -1,37 +1,95 @@
-import {LightningElement, api, track} from 'lwc';
+import { LightningElement, api, track } from 'lwc';
+import { NavigationMixin } from 'lightning/navigation';
 import GeFormService from 'c/geFormService';
+import { showToast, handleError } from 'c/utilTemplateBuilder';
 import DATA_IMPORT_BATCH_OBJECT from '@salesforce/schema/DataImportBatch__c';
+import DI_PAYMENT_AUTHORIZE_TOKEN_FIELD from '@salesforce/schema/DataImport__c.Payment_Authorization_Token__c';
+import DI_PAYMENT_STATUS_FIELD from '@salesforce/schema/DataImport__c.Payment_Status__c';
 
-export default class GeGiftEntryFormApp extends LightningElement {
+const PAYMENT_AUTHORIZE_TOKEN__C = DI_PAYMENT_AUTHORIZE_TOKEN_FIELD.fieldApiName;
+const PAYMENT_STATUS__C = DI_PAYMENT_STATUS_FIELD.fieldApiName;
+const CAPTURED = 'CAPTURED';
+
+export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement) {
     @api recordId;
     @api sObjectName;
 
     @track isPermissionError;
+    @track loadingText = 'Saving data import...';
+
+    get isBatchMode() {
+        return this.sObjectName &&
+            this.sObjectName === DATA_IMPORT_BATCH_OBJECT.objectApiName;
+    }
 
-    handleSubmit(event) {
+    handleBatchSubmit(event) {
         const table = this.template.querySelector('c-ge-batch-gift-entry-table');
-        const { diRecord, widgetValues } = event.detail.data;
-        let widgetString = '';
-        if(widgetValues) {
-            widgetString = JSON.stringify(widgetValues);
-        }
-        GeFormService.saveAndDryRun(
-            this.recordId, diRecord, widgetString)
-            .then(
-                dataImportModel => {
-                    Object.assign(dataImportModel.dataImportRows[0],
-                        dataImportModel.dataImportRows[0].record);
-                    table.upsertData(dataImportModel.dataImportRows[0], 'Id');
-                    table.setTotalCount(dataImportModel.totalCountOfRows);
-                    table.setTotalAmount(dataImportModel.totalAmountOfRows);
-                    event.detail.success(); //Re-enable the Save button
-                }
-            )
+        const dataImportRecord = event.detail.dataImportRecord;
+        GeFormService.saveAndDryRun(this.recordId, dataImportRecord)
```

Comment:
The three methods called in GeFormService (saveAndDryRun, handlePaymentProcessing, and handleProcessDataImport) are all only called from this component.  Could they each be imported by geGiftEntryFormApp directly so these call sites would reference the imported methods as opposed to calling methods on the GeFormService component's controller?

Reply:
Agree! Importing these directly into `geGiftEntryFormApp` now.